server/ladder_service.py

```diff
@@ -37,19 +40,102 @@ def __init__(
+                'ladder1v1',
+                map_pools=[(self.ladder_1v1_map_pool, None, None)]
+            )
         }
 
         self.searches: Dict[str, Dict[Player, Search]] = defaultdict(dict)
 
     async def initialize(self) -> None:
+        await self.update_data()
+        self._update_cron = aiocron.crontab('*/10 * * * *', func=self.update_data)
         await asyncio.gather(*[
             queue.initialize() for queue in self.queues.values()
         ])
         asyncio.create_task(self.handle_queue_matches())
 
+    async def update_data(self) -> None:
+        async with self._db.acquire() as conn:
+            # Legacy ladder1v1 map pool
+            result = await conn.execute(
+                "SELECT ladder_map.idmap, "
+                "table_map.name, "
+                "table_map.filename "
+                "FROM ladder_map "
+                "INNER JOIN table_map ON table_map.id = ladder_map.idmap"
+            )
+            maps = [
+                Map(row[0], row[1], row[2]) async for row in result
+            ]
+            self.ladder_1v1_map_pool.set_maps(maps)
+
+            # New map pools
+            result = await conn.execute(
+                select([
+                    map_pool.c.id,
+                    map_pool.c.name,
+                    map_version.c.map_id,
+                    map_version.c.filename,
+                    t_map.c.display_name
+                ]).select_from(
+                    map_pool.join(map_pool_map_version)
+                    .join(map_version)
+                    .join(t_map)
+                )
+            )
+            map_pool_maps = {}
+            async for row in result:
+                id_ = row[map_pool.c.id]
+                name = row[map_pool.c.name]
+                if id_ not in map_pool_maps:
+                    map_pool_maps[id_] = (name, list())
+                _, map_list = map_pool_maps[id_]
+                map_list.append(
+                    Map(
+                        row[map_version.c.map_id],
+                        row[t_map.c.display_name],
+                        row[map_version.c.filename]
+                    )
+                )
+
+            # Update the matchmaker queues
+            result = await conn.execute(
+                select([matchmaker_queue, matchmaker_queue_map_pool])
+                .select_from(matchmaker_queue.join(matchmaker_queue_map_pool))
+            )
+
+            queue_names = set()
+            async for row in result:
+                name = row[matchmaker_queue.c.technical_name]
+                if name not in self.queues:
+                    self.queues[name] = MatchmakerQueue(
+                        name=name,
+                        name_key=row[matchmaker_queue.c.name_key],
+                        game_service=self.game_service
+                    )
+                queue = self.queues[name]
+                if name not in queue_names:
+                    queue.map_pools.clear()
```

Comment:
I need to avoid destroying MatchmakerQueues to make sure the legacy one that's hardcoded doesn't get destroyed until there is actually a replacement for it in the database.